### PR TITLE
pakcs: 3.6.0 -> 3.7.2, build on ghc9

### PR DESCRIPTION
**Pull request by t4ccer**
version bump and switch to ghc9

cc #346214

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by maralorn** [comment]
I’d prefer not vendoring the patch but fetching it with fetchpatch from a PR against upstream, but its your decision.

**Comment by maralorn** [comment]
I seriously recommend that we activate the pakcs job in pkgs/top-level/release-haskell.nix (uncomment line 335) so that we notice when one of the haskell-updates breaks pakcs again.

File: pkgs/development/compilers/pakcs/default.nix [hunk 1]
```diff
@@ -1,25 +1,39 @@
...
-        inherit src;
-        postUnpack = "sourceRoot+=/frontend";
-      }) (super.callPackage ./curry-frontend.nix { });
+      curry-frontend = lib.pipe (super.callPackage ./curry-frontend.nix { })
+        [ haskell.lib.doJailbreak
+          (haskell.lib.compose.overrideCabal (drv: {
+            inherit src;
+            postUnpack = "sourceRoot+=/frontend";
+
+            # NOTE: We cannot use `appendPatches` because it conflicts with `overrideCabal`
```
Comment (sternenseemann) [inline pkgs/development/compilers/pakcs/default.nix:25]: `appendPatches` should do the same as you have written out here. Any specific problems?

Comment (t4ccer) [reply pkgs/development/compilers/pakcs/default.nix:25]: Oh ok `haskell.lib.compose.appendPatch` works, I've tried `haskell.lib.appendPatch` before and it doesn't compose well with `overrideCabal` I guess
```
error:
       … while calling the 'derivationStrict' builtin
         at <nix/derivation-internal.nix>:34:12:
           33|
           34|   strict = derivationStrict drvAttrs;
             |            ^
           35|

       … while evaluating derivation 'pakcs-3.7.2'
         whose name attribute is located at /home/t4ccer/repos/github/NixOS/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:336:7

       … while evaluating attribute 'makeFlags' of derivation 'pakcs-3.7.2'
         at /home/t4ccer/repos/github/NixOS/nixpkgs/pkgs/development/compilers/pakcs/default.nix:43:3:
           42|
           43|   makeFlags = [
             |   ^
           44|     "CURRYFRONTEND=${curry-frontend}/bin/curry-frontend"

       (stack trace truncated; use '--show-trace' to show the full, detailed trace)

       error: attribute 'override' missing
       at /home/t4ccer/repos/github/NixOS/nixpkgs/pkgs/development/haskell-modules/lib/compose.nix:38:28:
           37|    */
           38|   overrideCabal = f: drv: (drv.override (args: args // {
             |                            ^
           39|     mkDerivation = drv: (args.mkDerivation drv).override f;
           ```

Comment (t4ccer) [reply pkgs/development/compilers/pakcs/default.nix]: I've pushed this change, doesn't even trigger a rebuild

**Review by sternenseemann — APPROVED** [review]

**Pull request merged**